test/integration/custom-routes-i18n-index-redirect/next.config.js

```diff
@@ -0,0 +1,17 @@
+const destination = 'http://localhost:__EXTERNAL_PORT__'
+
+module.exports = {
+  i18n: {
+    locales: ['fr', 'en'],
+    defaultLocale: 'en',
+  },
+  async redirects() {
+    return [
+      {
+        source: '/',
+        destination: '/destination',
+        permanent: false,
+      },
+    ]
+  },
+}
```

Comment:
I ended up creating a new test directory `custom-routes-i18n-index-redirect`, based on `custom-routes-i18n`:

https://github.com/vercel/next.js/tree/7bbd239fe7e6d72ef467484619b978c11f5fbb70/test/integration/custom-routes-i18n

I did this instead of modifying `custom-routes-i18n` because `custom-routes-i18n` was testing `/` for `rewrites`, so it would conflict if I had modified it:

https://github.com/vercel/next.js/blob/7bbd239fe7e6d72ef467484619b978c11f5fbb70/test/integration/custom-routes-i18n/test/index.test.js#L55-L56